### ISSUE TITLE
Tables do not fit correctly in Word output

### ISSUE THREAD

**Issue by ian-curtis** [ian-curtis/gvsu215#56]
See #54 

Looks like the issue was lying behind some tables being fit before header labels being applied. Fitting should always be the last step, I've figured out. Code as of now has this built in, but it needs some testing to verify that each table works correctly with Word output.

@dilernia Would you be able to run a few tests on Word output? I did a little bit to make sure my new function `fit_tbl()` was working ok, but I did not do all tables as I kept getting a corrupt .rdb error (I think I have too many versions of `gvsu215` installed right now).

**Comment by ian-curtis** [comment]
Once we've confirmed this is functional, I'll add the new function information into the extended default behavior page.

**Comment by dilernia** [comment]
That's my understanding of flextable as well - subsequent layers, such as
adding captions, can "undo" fitting to the page (a bit surprising!). Glad
you got it squared away, nice work!

I reinstalled the package and tested it out on my end. The table functions
I tested out look better, although tbl_2var() and infer_2mean_test() do
have breaks in some columns still. I attached my .Rmd file for testing that
should run agnostic of file paths to show what I see.

Thanks!

Andrew

**Comment by ian-curtis** [comment]
Thank you!

I just pushed a super small change editing the max table width for Word output to be 6.9 inches (rather than 6.5). This fixed one line break issue I saw on my end. I don't see a .Rmd file, if you still see that style error, could you send it directly to me?

**Comment by ian-curtis** [comment]
Alright, I think the main issue here is with tables crossing page breaks. Widths look ok but if a large table crosses a page, the header repeats itself on the second page. Not necessarily a bad thing but could make a table look off.

Based on my research, `flextable::paginate()` (there are some arguments to it as well) solves the issue. I tested it on your .Rmd file and it worked great.

```
tbl_2var(data = Executions, Method ~ Race, row_pct = "show", digits = 3, 
         caption = NULL) |>
  flextable::paginate()
```

The main question is, should this be built into the package or performed by the user as needed? I'm leaning towards the latter as not all tables in all instances will require a page break and it wouldn't be a good idea to add a page break before every table. I'm thinking about the following:

1. I do nothing and let the user supply the `paginate()` function themselves as needed.
2. I build in `paginate()` to all tables that are likely to be long.
3. I build in a check to call `paginate()` conditionally, such as after a certain number of rows in the table.
4. I build in a `break_page = c("no", "yes")` argument to every table to give a user the option each time.

What do you think? Again, I worry about automating this as these kinds of page breaks depend entirely on the data used, the text printed on the Word doc, and how much space plots and other tables before/after take up.

Adding @dykesb here for thoughts.

**Comment by dilernia** [comment]
I don't feel there's any issue with the default behavior of repeating column names in the table after a page break personally. My only concern with the tables was line breaks in column names within a table (unrelated to page breaks), but if that issue is resolved then I think we should be all set so option 1 sounds best to me.

That paginate() function is great to know though! 

Separately, I was replying to these via email which I think is why my attachments weren't getting through, but now I see I'll reply on GitHub moving forward.

**Comment by ian-curtis** [comment]
That works! I looked at that doc you sent again and tested my changes. It looks like line breaks in the table are good to go now (I forced line breaks in any two+ word label name). The only way that it might not work well is if the variable names themselves are long (which I can't control). The table is set to fit to the maximum width it can without bleeding into margins so I think any further adjustments would have to be done by the user before or after making the table. 

I'll close this issue with the next pull request which should publish the corrections here and likely bump us to 1.0!
